_javascripts/page-loader.js

```diff
@@ -1,76 +1,40 @@
-// the new final link to load
-newLink = "";
-newVersion = "";
-currentVersion = "";
-
-// the fileRequested
-fileRequested = "";
-
-//Array prototype to check if an element is in an array
-Array.prototype.contains = function(obj) {
-  return this.indexOf(obj) > -1;
-}
+let newLink = "";
+let newVersion = "";
+let currentVersion = "";
+let fileRequested = "";
+
+const urlMappings = {
+  "openshift-acs": "https://docs.openshift.com/acs/",
+  "openshift-builds": "https://docs.openshift.com/builds/",
+  "openshift-enterprise": "https://docs.openshift.com/container-platform/",
+  "openshift-gitops": "https://docs.openshift.com/gitops/",
+  "openshift-origin": "https://docs.okd.io/",
+  "openshift-pipelines": "https://docs.openshift.com/pipelines/",
+  "openshift-serverless": "https://docs.openshift.com/serverless/",
+};
 
 function versionSelector(list) {
+  "use strict";
 
-  // the version we want
   newVersion = list[list.selectedIndex].value;
+  currentVersion = window.location.pathname.split("/")[2];
 
-  // spilt the current path
-  var pathArray = window.location.pathname.split( '/' );
-
-  // so we can get the current version
-  currentVersion = pathArray[2];
+  let baseUrl = urlMappings[dk];
 
-  // if switching major versions, just take the user to the main landing page
-  // as files change a lot between major versions.
-
-  if(currentVersion.charAt(0) === newVersion.charAt(0)) {
-    // the file path is just the version number + the end of the path
-    fileRequested =
-      window.location.pathname.substring(
-        window.location.pathname.lastIndexOf(currentVersion) +
-        currentVersion.length);
-  } else {
-    fileRequested = "/welcome/index.html";
+  //Handle special OCP case
+  if (["3.0", "3.1", "3.2"].includes(newVersion) && dk === "openshift-enterprise") {
+    baseUrl = "https://docs.openshift.com/enterprise/";
   }
 
-  // alert(fileRequested);
-
-  // in 3.3 and above, we changed to container-platform
-  if(['3.0', '3.1', '3.2'].contains(newVersion)) {
-    newLink = "https://docs.openshift.com/enterprise/" +
-      newVersion +
-      fileRequested;
-  } else if (['3.65', '3.66', '3.67', '3.68', '3.69', '3.70', '3.71', '3.72', '3.73', '3.74', '4.0', '4.1'].contains(newVersion)) {
-    // check and handle links for RHACS versions
-    newLink = "https://docs.openshift.com/acs/" +
-      newVersion +
-      fileRequested;
-  } else if (['1.28', '1.29'].contains(newVersion)) {
-    // check and handle links for Serverless versions
-    newLink = "https://docs.openshift.com/serverless/" +
-      newVersion +
-      fileRequested;
-  } else if (['1.8', '1.9'].contains(newVersion)) {
-    // check and handle links for GitOps versions
-    newLink = "https://docs.openshift.com/gitops/" +
-      newVersion +
-      fileRequested;
+  if (dk === "openshift-enterprise" && currentVersion.charAt(0) !== newVersion.charAt(0)){
+    fileRequested = "/welcome/index.html";
```

Comment:
Not all distros have a welcome/index.html, eg., https://github.com/openshift/openshift-docs/tree/serverless-docs-1.28

Reply:
Yes, this only is for OpenShift enterprise.